_sources/structs/mixed_up_code.rst

```diff
@@ -0,0 +1,236 @@
+Mixed Up Code Exercises
+-----------------------
+
+These questions are **completely OPTIONAL**. We provide these exercises to give you more practice with the material that you've learned. 
+Additionally, they may be a helpful resource when reviewing for assessments. Some lines contain **mistakes** or are **unnecessary** for 
+the function - these lines should not be selected. Make sure to place the blocks at the right indentation levels!
+
+.. parsonsprob:: structs_mixed_up_code_1
+    :numbered: left
+    :adaptive:
+
+    Let's write the code for the struct definition of ``Song``. 
+    The Song structure will have the instance variables ``string title``, 
+    ``string artist``, ``string album``, and ``int year`` in that order. 
+    Put the necessary blocks of code in the correct order.
+    -----
+    struct Song {
+    =====
+    struct song { #paired
+    =====
+        string title;
+    =====
+        string artist;
+    =====
+        string album;
+    =====
+        int year;
+    =====
+        string year;  #distractor
+    =====
+    };
+    =====
+    } #distractor
+
+.. parsonsprob:: structs_mixed_up_code_2
+    :numbered: left
+    :adaptive:
+
+    Let's write the code for the ``printSong`` function. ``printSong``
+    takes a ``Song`` as a parameter and prints out the instance variables
+    in the following format: ``"title" by artist (album, year)``. Put the necessary blocks of 
+    code in the correct order.
+    -----
+    void printSong (Song &s) {
+    =====
+        cout << "\"" << s.title << "\" by " << s.artist;
+    =====
+        cout << " (" << s.album << ", " << s.year << ")" << endl;
+    =====
+        cout << title << artist << album << year;  #distractor
+    =====
+        cout << "\"" << title << "\" by " << artist;  #distractor
+    =====
+    }
+
+.. parsonsprob:: structs_mixed_up_code_3
+    :numbered: left
+    :adaptive:
+
+    Let's write the code for the struct definition of ``Unicorn``. 
+    The ``Unicorn`` structure will have the instance variables ``name``, 
+    ``age``, ``hornLength``, ``hairColor``, and ``isSparkly`` in that order. A Unicorn's
+    ``hornLength`` is measured to the nearest tenth of a unit and ``isSparkly`` is a boolean.
+    Put the necessary blocks of code in the correct order.
+    -----
+    struct Unicorn {
+    =====
+    Struct Unicorn {  #distractor
+    =====
+        string name;
+    =====
+        int age;
+    =====
+        double hornLength;
+    =====
+        string hairColor;
+    =====
+        bool isSparkly;
+    =====
+        int hornLength;  #distractor
+    =====
+    };
+    =====
+    } #distractor
+
+.. parsonsprob:: structs_mixed_up_code_4
+    :numbered: left
+    :adaptive:
+
+    Let's write the code for the ``convertToHumanAge`` function. ``convertToHumanAge``
```

Comment:
This one is doing that weird thing where I can see the question, but the Parson's problem doesn't have any code blocks in it. I'm assuming you were trying to delete this question?

Reply:
All of the questions that you noted were not showing up correctly in the structs chapter and vectors chapter now have updated ID's, they should now display on the page with all of the correct blocks along with the question text.

Reply:
Thanks! They show up now.